tests/console/validate_mirror_repos.pm

```diff
@@ -0,0 +1,46 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate that mirror used for installation is added as a repo in the installed system.
+#
+# Maintainer: Joaquín Rivera <jeriveramoya@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+
+sub run {
+    select_console 'root-console';
+
+    # extract string with medium after last slash
+    my $medium_extraction = qr/(?<medium>[^\/]+)$/;
+
+    # get medium from settings
+    my $method     = uc get_required_var('INSTALL_SOURCE');
+    my $mirror_src = get_required_var("MIRROR_$method");
+    $mirror_src =~ $medium_extraction;
+    my $medium = $+{medium};
+
+    # parse output of zypper to compare medium from settings with medium from URIs
+    my @uris = split(/\n/, script_output("zypper lr --uri | awk -F '|' '{print \$7}'"));
+    my @wrong_uris;
+    foreach my $uri (@uris) {
```

Comment:
Ok, I guess this is where we have different point of views. So you want to check all the repos being added to the system, which is more like testing proxy SCC, as they come from it.
I would say that we need to check exact url which we have used for the installation and that repo also should be disabled AFAIR. For other repos, we can check that for each enabled module, we got the repo, and this check will be relevant for any installation using Online medium.